packages/core/src/queries/user.ts

```diff
@@ -34,10 +36,22 @@ export const findUserByPhone = async (phone: string) =>
   `);
 
 export const findUserById = async (id: string) =>
-  envSet.pool.one<User>(sql`
-    select ${sql.join(Object.values(fields), sql`,`)}
+  envSet.pool.one<UserWithRoleNames>(sql`
+    select ${sql.join(
+      Object.values(fields).map((field) => sql`${table}.${field}`),
+      sql`,`
+    )}, coalesce(json_agg(${rolesTable}.${rolesFields.name}) filter (where ${rolesTable}.${
+    rolesFields.name
+  } is not null), '[]') as role_names
     from ${table}
-    where ${fields.id}=${id}
+    left join ${usersRolesTable} ON ${table}.${fields.id} = ${usersRolesTable}.${
+    usersRolesFields.userId
+  }
+    left join ${rolesTable} ON ${rolesTable}.${rolesFields.id} = ${usersRolesTable}.${
+    usersRolesFields.roleId
+  }
+    where ${table}.${fields.id}=${id}
+    group by ${table}.${fields.id}
```

Comment:
instead of joining all things here, how about we do 2 queries:
- findUserByIds() // can be multiple or large query
- findRoles() // number of roles are limited
and merge them somewhere. (either backend or frontend)

Reply:
But there is still a "relation" table for "user" and "role", get "roles" first won't enough, we still need the data in "users_roles" which may be large.

Reply:
But 3 steps can avoid join:

- findUserByIds
- findUserRoles
- findRoles

Reply:
updated